### PR TITLE
Add plugin: Vitepress Publisher

### PR DESCRIPTION
**Pull request by tyrad**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by liamcain** [comment]
- [this.settingTab = new SettingTab(this.app, this)](https://github.com/tyrad/obsidian-vitepress/blob/35ed82fb88f4a43ea4550655aa2be39e4717e481/src/main.ts#L19) You shouldn't need a reference to this. Just create the setting tab inside onload
- [button.detach();](https://github.com/tyrad/obsidian-vitepress/blob/35ed82fb88f4a43ea4550655aa2be39e4717e481/src/main.ts#L98) why are you detaching the existing button? If the button already exists, you should just skip. Otherwise you'll be mutating the dom more than you need to.
- [setTimeout](https://github.com/tyrad/obsidian-vitepress/blob/35ed82fb88f4a43ea4550655aa2be39e4717e481/src/main.ts#L111) what's this setTimout for?
- [activeLeaf](https://github.com/tyrad/obsidian-vitepress/blob/35ed82fb88f4a43ea4550655aa2be39e4717e481/src/main.ts#L90) maybe check the view type of the leaf too
- [const activeLeaf](https://github.com/tyrad/obsidian-vitepress/blob/35ed82fb88f4a43ea4550655aa2be39e4717e481/src/main.ts#L90) this name is not accurate. it's not the active leaf.
